### PR TITLE
Initial Cloud Foundry documentation

### PR DESCRIPTION
**Pull request by ryanjbaxter**

### CODE REVIEW

**Comment by ryanjbaxter** [comment]
@codydaig here is my first stab at moving the documentation for Cloud Foundry.  I tried to run `jekyll serve` to see what it looks like but got an error
```
$ jekyll serve
Configuration file: /Users/ryanjbaxter/git-repos/ryanjbaxter/mean/_config.yml
            Source: /Users/ryanjbaxter/git-repos/ryanjbaxter/mean
       Destination: /Users/ryanjbaxter/git-repos/ryanjbaxter/mean/_site
      Generating...
jekyll 2.5.3 | Error:  undefined method `default_proc=' for "this --- is correct":String
```
any idea how to fix that?

**Comment by codydaig** [comment]
@ryanjbaxter Ah. Yes. That's a funky issue. If you remove the node_modules folder from the root directory it will work. I will submit a pull request to fix that and have jekyll exclude that directory. Because node_modules is in the gitignore, when you switch branches, it leaves it behind.

**Comment by ryanjbaxter** [comment]
@codydaig thanks that worked!  I made some more minor tweaks and added links to the CF documentation from the main page.  Let me know what you think.

**Comment by codydaig** [comment]
@ryanjbaxter The final structure change just got merged in, so the links that you added and going to cause conflicts that need to get resolved before this can be merged in.

**Comment by lirantal** [comment]
Guys we can't merge this because of conflicts, please take a look.
Also @ryanjbaxter please rebase and squash your commits so we can merge.

**Comment by ryanjbaxter** [comment]
@codydaig and @lirantal i just fixed the merge conflicts

**Comment by codydaig** [comment]
@ryanjbaxter You have docs04x.html in the root directory and that file does not exist anymore. Delete that file and then it will be ready to merge.

**Comment by ilanbiala** [comment]
@codydaig is everything good other than the extra html file? @lirantal are you familiar with CF at all? I'm not really familiar.

**Comment by lirantal** [comment]
@ilanbiala nope.

**Comment by codydaig** [comment]
@ilanbiala I'm not familiar with CF either. I'd suggest we push this off to 0.4.x, then I can test within the next week before we merge this. Plus, this requires #681 to be merged in too be valid.

**Comment by ilanbiala** [comment]
@lirantal @codydaig has a good idea. Let's push it off.

**Comment by lirantal** [comment]
Sure, it's nothing critical for 0.4.0

**Comment by ryanjbaxter** [comment]
@lirantal @ilanbiala @codydaig this should now be all set. as should be #732 

Not sure why it needs to be pushed off now, the instructions explain how to test this out and I am willing to help anyone who needs it.

**Comment by ryanjbaxter** [comment]
My initial pull request was pushed off for months I dont want this to happen again.

**Comment by lirantal** [comment]
@ryanjbaxter understood, don't worry, we're just getting through 0.4.0 first.

**Comment by ilanbiala** [comment]
@ryanjbaxter it doesn't even need to be pushed off. Since it isn't breaking anything, we can just release 0.4.1 with this change once we get it tested.

**Comment by ryanjbaxter** [comment]
@ilanbiala ok as long as that happens fairly quickly

**Comment by codydaig** [comment]
@ryanjbaxter Having experience in dev ops, how long do you think it will take me to setup?

**Comment by ryanjbaxter** [comment]
@codydaig it will take you a minute to sign up for Bluemix or Pivotal Web Services, 3 minutes to install the CF CLI, and then you can deploy the app.  If you want to setup the social services you will spend a couple minutes issuing some basic CF CLI commands, nothing hard though.

**Comment by codydaig** [comment]
@ryanjbaxter Here's my command line history showing you the steps that I took. The node_modules folder got uploaded with the cf push and the app is not working.
http://mean-exosporous-ensemble.cfapps.io/
```
522  cf login -a https://api.run.pivotal.io
523  cd ~/Desktop/mean-0.4.0-cf
524  cf create-service mongolab sandbox mean-mongo
525  npm install
526  cf push
```

**Comment by ryanjbaxter** [comment]
@codydaig thanks for trying it out!  I screwed up the formatting in the .cfignore file, it should be fixed now (as well as the documentation issues) can you try again?

**Comment by codydaig** [comment]
@ryanjbaxter I'm probably overlooking something, but if you are excluding node_modules and public/lib, why do we need to run npm install and grunt build locally before we push? (other than local dev reasons)

**Comment by ryanjbaxter** [comment]
@codydaig because when deployed to CF we use the JS in public/dist

**Comment by codydaig** [comment]
Yup. Knew I was overlooking something small. 
It worked! http://mean-unfacaded-desponder.cfapps.io/
Side question, anytime I make a change and push, it adds another url to the urls list. So currently, this app has 3 urls that all render the same and access the same database. Is this normal or what am I missing? (I'm on gitter if that's easier too)

**Comment by codydaig** [comment]
@ryanjbaxter - Per the conversation on Gitter, Add something to the documentation to change the app name in the manifest file and the documentation for this will be good to go. (Grammar is not my strong suite, so if anyone wants to check grammar, they should do so).

**Comment by ryanjbaxter** [comment]
@codydaig I added a troubleshooting section specifying how to deal with the multiple urls.

There is also one more enhancement I would like to make to the documentation to make deployments easier but it can be (and had to be made) after these changes are merged.  I can add a "Deploy To Bluemix" button to the documentation that allows the user to deploy MEANJS to Bluemix with a single click.  This way they dont have to install the CF CLI, run npm, and run grunt to get it deployed.  However I need the changes in #732 to be merged in before testing this.

**Comment by codydaig** [comment]
@ryanjbaxter Congrats! Your commit has been merged in! Is this ready to be reviewed and merged in or do you want to make your change with the one click button?

File: _includes/docs/0.4.x/cloud-foundry.html [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        `cf push` commands Cloud Foundry will bind a new random URL or route to your application each time.
+        This means you end up with multiple URLs pointing to the same application instance.  While this does not hurt
+        your application it is generally not necessary.  You can fix this by changing the host property in
+        your `manifest.yml` file to remove the `${random-word}` string from it.  However note that if your
+        host name is not unique, your deployment will fail.  If it does fail try changing the host property to
+        something else.
+      </ul>
+    </li>
+  </ul>
+</section>
```
Comment (codydaig) [inline _includes/docs/0.4.x/cloud-foundry.html]: @ryanjbaxter New Line Please

**Comment by codydaig** [comment]
@ryanjbaxter Go ahead and squash the commits and add the new line and I'll give my approval.

**Comment by codydaig** [comment]
LGTM

**Comment by lirantal** [comment]
Great, thanks @ryanjbaxter !

**Pull request merged**